src/modules/rlm_eap/libeap/mppe_keys.c

```diff
@@ -301,9 +301,17 @@ void eap_fast_tls_gen_challenge(SSL *s, uint8_t *buffer, uint8_t *scratch, size_
 	p = seed;
 	memcpy(p, prf_label, len);
 	p += len;
+#ifdef HAVE_SSL_GET_SERVER_RANDOM
+	SSL_get_server_random(s, p, SSL3_RANDOM_SIZE);
+#else
 	memcpy(p, s->s3->server_random, SSL3_RANDOM_SIZE);
+#endif
 	p += SSL3_RANDOM_SIZE;
+#ifdef HAVE_SSL_GET_CLIENT_RANDOM
```

Comment:
I'd prefer to just use the local version of `SSL_get_server_random()` instead of `ifdef`s.  We took a pass through the code to do this, but haven't finished it yet.